### PR TITLE
Fix almost all the pyproject.toml and poetry.lock files to have consistent requirements on python dependencies.

### PR DESCRIPTION
**Pull request by eric-anderson**
demo-ui can't be fixed as it has conflicting requirements.
add script to perform the work of cross-checking.
manually resolve (and document why) all the special cases in the new lib/poetry-lock "library"

### CODE REVIEW

File: lib/poetry-lock/poetry-lock-all.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+for i in **/pyproject.toml; do
+    # Demo UI needs 0.28, rest of sycamore needs 1.x
+    [[ $i = *openai-proxy* ]] && continue
+    (
+        echo "--------------------- proecessing in $i"
+        cd $(dirname "$i")
+        poetry lock --no-update || return 1
+        poetry install |& tee /tmp/poetry-install.out || return 1
+        perl -ne 'print qq{$1 = "$2"\n} if /Downgrading (\S+) \((\S+) ->/o;' </tmp/poetry-install.out
+    )
```
Comment (HenryL27) [inline lib/poetry-lock/poetry-lock-all.sh:30]: Ok I think I understand that this is the 'lock everything' section but I have no idea what this is actually doing. also nit: proecessing -> processing

File: lib/poetry-lock/sycamore_poetry_lock/noop.py [hunk 2]
```diff
@@ -0,0 +1 @@
+pass
```
Comment (HenryL27) [inline lib/poetry-lock/sycamore_poetry_lock/noop.py:1]: nice

**Review by HenryL27 — APPROVED** [review]
Cool!

**Comment by HenryL27** [comment]
Although won't this leave us no hope of reducing our container sizes?

**Comment by eric-anderson** [comment]
This shouldn't affect the container sizes. When we do installs we don't install the dev dependencies.

**Pull request merged**